lib/CL/devices/level0/pocl-level0.cc

```diff
@@ -354,6 +353,11 @@ unsigned int pocl_level0_probe(struct pocl_device_ops *Ops) {
     return 0;
   }
 
+  int EnvCount = pocl_device_get_env_count(Ops->device_name);
+  if (EnvCount > 0 && EnvCount < DriverCount) {
+    DriverCount = EnvCount;
+  }
+
```

Comment:
If I understood the changes in this function correctly, this will enable level0 device when `POCL_DEVICES` is unset (1) or it is set to something that does not have occurrence of "level0" (2) - is that intentional? This by assuming there are level0 devices on the system.

`pocl_device_get_env_count` returns -1 for the case (1) and 0 for the case (2). In both cases `pocl_level0_probe` returns total number of devices available but previously it returned 0.

Reply:
I've changed it to match the behavior of CPU driver: case 1 (unset) will enable all autodetected LZ devices, case 2 there will be zero LZ devices reported.